packages/components/src/list/experimental-list-item.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { CSSTransition } from 'react-transition-group';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { handleKeyDown } from './list-item';
+
+type CSSTransitionProps = {
+	in: boolean;
+	exit: boolean;
+	enter: boolean;
+	onExited: () => void;
+};
+
+type ListItemProps = {
+	// control whether to display padding on list item or not.
+	disableGutters: boolean;
+	className?: string;
+	// By default the List is a `nav`, which treats the list item as role `link` if `onClick` is passed. To override this pass a different role.
+	role?: 'menuitem' | 'link' | 'button';
+	// By default a div is rendered, but if you want the list item to behave as a different tag you can override it here.
+	component?: React.ElementType;
+} & CSSTransitionProps &
+	React.AllHTMLAttributes< HTMLElement >;
+
+export const ExperimentalListItem: React.FC< ListItemProps > = ( {
+	children,
+	disableGutters = false,
+	className = '',
+	component = 'a',
+	tabIndex,
+	// extract out the props that must be passed down from TransitionGroup
+	exit,
+	enter,
+	onExited,
+	// in is a TS reserved keyword so can't be a variable name
+	in: transitionIn,
+
+	// Everything else you might pass into an HTML element
+	...otherProps
+} ): JSX.Element => {
+	// for styling purposes only
```

Comment:
@psealock @mattsherman wondering if you could take another look. Looking back on my first approach it was pretty naive, some of the assumptions made no sense based on how I implemented the `<List>`

Let me talk through reasoning of what I have now.

I originally made the default component for List a `nav` to satisfy the original problem of the list not being semantically correct or satisfying aria concerns raised by @mattsherman , but then I realised thanks to @psealock 's comments that all the work I was doing in the list item, was circumventing it acting like a link, and there was convoluted logic to add roles. It was silly on retrospect, if the default is a link list, then let's default to an actual link! roles after all are a workaround for when you *can't* use the semantically correct markup. But here we totally can.

So the approach is now: List is a `nav`, list items are `a` tags. no roles needed since its all semantically correct. link previews work. 

If you want to introduce a different semantic approach you can do that by overriding `component` in List and ListItem then you can do that. If you provide an `onClick` handler and a different `component` then it'll not show a link preview but it will have a `cursor: pointer`, so its purely presentational, providing the right roles will be up to the implementer.

Keen to hear your thoughts! One other upside I can think of is, that it does satisfy the core use case within wc-admin but without painting anyone else using it into a corner either.

Reply:
Also worth noting I never bothered with deciding whether to render a `button`, that one thing feels like the wrong approach to me. Button has a lot of styling associated with it which is a pain, I think we should leave it up to the implementer if they want to use a button or whether they want to add a role to a div.